skpro/regression/cyclic_boosting.py

```diff
@@ -0,0 +1,315 @@
+        for i in interaction:
+            features.append(i)
+
+        for quantile in self.quantiles:
+            self.quantile_est.append(
+                pipeline_CBMultiplicativeQuantileRegressor(
+                    quantile=quantile,
+                    feature_properties=self.feature_properties,
+                    feature_groups=features,
+                    maximal_iterations=50,
+                )
+            )
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+
+        self._y_cols = y.columns
+        y = y.to_numpy().flatten()
+
+        # multiple quantile regression for full probability estimation
+        for est in self.quantile_est:
+            est.fit(X.copy(), y)
+
+        return self
+
+    def _predict(self, X):
+        """Predict labels for data from features.
+
+        State required:
+            Requires state to be "fitted" = self.is_fitted=True
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : pandas DataFrame, same length as `X`, same columns as `y` in `fit`
+            labels predicted for `X`
+        """
+
+        index = X.index
+        y_cols = self._y_cols
+
+        # quantile prediction
+        for est in self.quantile_est:
+            yhat = est.predict(X.copy())
+            self.quantile_values.append(yhat)
+
+        j_qpd = J_QPD_S(
+            0.2,
+            qv_low=self.quantile_values[0],
+            qv_median=self.quantile_values[1],
+            qv_high=self.quantile_values[2],
+            index=index,
+            columns=y_cols,
+        )
+        self.qpd = j_qpd
+
+        y_pred = pd.DataFrame(j_qpd.mean(), index=index, columns=y_cols)
+        return y_pred
+
+    def _predict_proba(self, X):
+        """Predict distribution over labels for data from features.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y_pred : skpro BaseDistribution, same length as `X`
+            labels predicted for `X`
+        """
+
+        # predict j-qpd
+        _ = self._predict(X)
+
+        if self.distr_type == "QPD":
+            return self.qpd
+        else:
```

Comment:
I don't think we should offer this option, matching the QPD mean and variance on Gauss or Laplace distributions is odd at best (and can be severely off).

Reply:
up to you

Reply:
I removed these options